subsys/bluetooth/controller/ll_sw/ull_adv_sync.c

```diff
@@ -56,170 +59,234 @@ static void ticker_op_cb(uint32_t status, void *param);
+	if (err) {
+		return err;
 	}
 
-	/* TODO: ACAD */
-
-	/* TODO: AdvData */
-
-	/* Calc tertiary PDU len */
-	ter_len = ull_adv_aux_hdr_len_calc(ter_com_hdr, &ter_dptr);
-	ull_adv_aux_hdr_len_fill(ter_com_hdr, ter_len);
-
-	ter_pdu->len = ter_len;
-
+#if IS_ENABLED(CONFIG_BT_CTLR_DF_ADV_CTE_TX)
+	if (extra_data) {
+		adv_sync_extra_data_set_clear(extra_data_prev, extra_data,
+					      hdr_add_fields, hdr_rem_fields,
+					      (data ? data->extra_data : NULL));
+	}
+#endif /* CONFIG_BT_CTLR_DF_ADV_CTE_TX */
 	return 0;
 }
 
-uint8_t ll_adv_sync_ad_data_set(uint8_t handle, uint8_t op, uint8_t len,
-				uint8_t const *const data)
+#if IS_ENABLED(CONFIG_BT_CTLR_DF_ADV_CTE_TX)
+/* @brief Set or clear fields in extended advertising header and store
+ *        extra_data if requested.
+ *
+ * @param[in]  extra_data_prev  Pointer to previous content of extra_data.
+ * @param[in]  hdr_add_fields   Flag with information which fields add.
+ * @param[in]  hdr_rem_fields   Flag with information which fields remove.
+ * @param[in]  data             Pointer to data to be stored in extra_data.
+ *                              Content depends on the data depends on
+ *                              @p hdr_add_fields.
+ *
+ * @Note
+ * @p data depends on the flag provided by @p hdr_add_fields.
+ * Information about content of value may be found in description of
+ * @ref ull_adv_sync_pdu_set_clear.
+ *
+ * @return Zero in case of success, other value in case of failure.
+ */
+static inline void adv_sync_extra_data_set_clear(void *extra_data_prev,
+						 void *extra_data_new,
+						 uint16_t hdr_add_fields,
+						 uint16_t hdr_rem_fields,
+						 void *data)
+{
+	/* Currently only CTE enable requires extra_data. Due to that fact
+	 * CTE additional data are just copied to extra_data memory.
+	 */
+	if (hdr_add_fields & ULL_ADV_PDU_HDR_FIELD_CTE_INFO) {
+		memcpy(extra_data_new, data, sizeof(struct lll_df_adv_cfg));
+	} else if ((hdr_rem_fields & ULL_ADV_PDU_HDR_FIELD_CTE_INFO) ||
+		    extra_data_prev) {
+		memmove(extra_data_new, extra_data_prev,
+			sizeof(struct lll_df_adv_cfg));
+	}
+}
+#endif /* CONFIG_BT_CTLR_DF_ADV_CTE_TX */
+
+/* @brief Set or clear fields in extended advertising header.
+ *
+ * @param[in]  adv              Advertising set.
+ * @param[in]  hdr_add_fields   Flag with information which fields add.
+ * @param[in]  hdr_rem_fields   Flag with information which fields remove.
+ * @param[in]  value            Pointer to data to be added to header.
+ *                              Content depends on the value of
+ *                              @p hdr_add_fields.
+ * @param[out] ter_idx          Index of new PDU.
+ *
+ * @Note
+ * @p value depends on the flag provided by @p hdr_add_fields.
+ * Information about content of value may be found in description of
+ * @ref ull_adv_sync_pdu_set_clear.
+ *
+ * @return Zero in case of success, other value in case of failure.
+ */
+static uint8_t adv_sync_hdr_set_clear(struct lll_adv_sync *lll_sync,
```

Comment:
Move this static function after the definition of `sync_remove` in this file.

Reply:
Code moves according to suggestion